metals/src/main/scala/scala/meta/internal/metals/Indexer.scala

```diff
@@ -117,6 +117,104 @@ case class Indexer(indexProviders: IndexProviders)(implicit rc: ReportContext) {
           importedBuild.dependencyModules
         )
 
+        // hack for mill build sources until a proper BSP wrapped sources mapping gets into
+        // the official protocol
+        val millBuildTargetIdentifiers =
+          importedBuild.workspaceBuildTargets
+            .getTargets()
+            .iterator()
+            .asScala
+            .collect {
+              case t if t.getName().endsWith("mill-build/") =>
```

Comment:
```suggestion
              case t if t.getName().endsWith("mill-build-") =>
```
weirdly it's this way for Bloop. With Mill BSP it works, but it seems BSP breaks on newest mill :sigh:

Reply:
This code is not production-ready yet. I was hacking it to make source mapping work, but I will leave it to another PR as you suggested.